auto_tune_vllm/core/study_controller.py

```
@@ -386,7 +386,16 @@ def run_optimization(
         total_trials = n_trials or self.config.optimization.n_trials
-        max_concurrent = max_concurrent or float('inf')
+        
+        # Require explicit concurrency specification
+        if max_concurrent is None:
+            raise ValueError(
+                "❌ --max-concurrent is required to prevent GPU memory conflicts!\n\n"
+                "Add to your YAML config:\n"
+                "  optimization:\n"
+                "    max_concurrent: 2  # Match your GPU count\n\n"
+                "Or use CLI: --max-concurrent 2"
+            )
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)